doc-site/docs/tutorials/custom_contracts/cardano.md

```diff
@@ -0,0 +1,542 @@
+        "description": "",
+        "params": [
+          {
+            "name": "transactionId",
+            "schema": {
+              "type": "string"
+            }
+          }
+        ]
+      },
+      "location": {
+        "address": "sample-contract@0.1.0"
+      },
+      "interface": {
+        "id": "120d061e-bcda-4c2f-a296-018d7cd62a04"
+      },
+      "signature": "sample-contract@0.1.0:TransactionFinalized(string)"
+    }
+  ]
+}
+```
+
+We can see in the response, that FireFly pulls all the schema information from the FireFly Interface that we broadcasted earlier and creates the listener with that schema. This is useful so that we don't have to enter all of that data again.
+
+## Subscribe to events from our contract
+
+Now that we've told FireFly that it should listen for specific events on the blockchain, we can set up a **Subscription** for FireFly to send events to our app. To set up our subscription, we will make a `POST` to the `/subscriptions` endpoint.
+
+We will set a friendly name `sample-contract` to identify the Subscription when we are connecting to it in the next step.
+
+We're also going to set up a filter to only send events blockchain events from our listener that we created in the previous step. To do that, we'll **copy the listener ID** from the step above (`b314d8af-2641-4bf2-b386-2e658f3e76a5`) and set that as the value of the `listener` field in the example below:
+
+### Request
+
+`POST` `http://localhost:5000/api/v1/namespaces/default/subscriptions`
+
+```json
+{
+  "namespace": "default",
+  "name": "sample-contract",
+  "transport": "websockets",
+  "filter": {
+    "events": "blockchain_event_received",
+    "blockchainevent": {
+      "listener": "b314d8af-2641-4bf2-b386-2e658f3e76a5"
+    }
+  },
+  "options": {
+    "firstEvent": "oldest"
+  }
+}
+```
+
+### Response
+
+```json
+{
+  "id": "f826269c-65ed-4634-b24c-4f399ec53a32",
+  "namespace": "default",
+  "name": "sample-contract",
+  "transport": "websockets",
+  "filter": {
+    "events": "blockchain_event_received",
+    "message": {},
+    "transaction": {},
+    "blockchainevent": {
+      "listener": "b314d8af-2641-4bf2-b386-2e658f3e76a5"
+    }
+  },
+  "options": {
+    "firstEvent": "-1",
+    "withData": false
+  },
+  "created": "2025-03-15T17:35:30.131698921Z",
+  "updated": null
+}
+```
+
+## Receive custom smart contract events
+
+The last step is to connect a WebSocket client to FireFly to receive the event. You can use any WebSocket client you like, such as [Postman](https://www.postman.com/) or a command line app like [`websocat`](https://github.com/vi/websocat).
+
+Connect your WebSocket client to `ws://localhost:5000/ws`.
+
+After connecting the WebSocket client, send a message to tell FireFly to:
+
+- Start sending events
+- For the Subscription named `sample-contract`
+- On the `default` namespace
+- Automatically "ack" each event which will let FireFly immediately send the next event when available
```

Comment:
We recommend not using autoack as it's safer for application to manual ack once they have processed the event

Reply:
That's what the other contract readmes say to do. Are there instructions for how to consume this kind of API without autoack somewhere?

Reply:
Yes, there should be in the subscription section - let me pull it out

Reply:
So this one https://hyperledger.github.io/firefly/latest/reference/types/subscription/#using-start-and-ack-explicitly

Reply:
It does need some cleaning up maybe

Reply:
I added a note about autoack below this. I do think autoack behavior makes sense for initial setup/experimentation, as long as users know not to leave it on.